examples/c/profile.c

```diff
@@ -0,0 +1,209 @@
+// SPDX-License-Identifier: (LGPL-2.1 OR BSD-2-Clause)
+/* Copyright (c) 2022 Facebook */
+#include <stdio.h>
+#include <unistd.h>
+#include <string.h>
+#include <sys/syscall.h>
+#include <sys/sysinfo.h>
+#include <linux/perf_event.h>
+#include <bpf/libbpf.h>
+#include <bpf/bpf.h>
+
+#include "profile.skel.h"
+#include "profile.h"
+#include "blazesym.h"
+
+
+static long
+perf_event_open(struct perf_event_attr *hw_event, pid_t pid,
+		int cpu, int group_fd, unsigned long flags)
+{
+	int ret;
+
+	ret = syscall(__NR_perf_event_open, hw_event, pid, cpu,
+		      group_fd, flags);
+	return ret;
+}
+
+struct handler_ctx {
+	int stackmap_fd;
+	blazesym *symbolizer;
+	sym_file_cfg cfgs[2];
+};
+
+void
+show_stack_trace(struct handler_ctx *ctx, uint64_t *stack)
+{
+	const struct blazesym_result *syms;
+	int cnt, i;
+
+	for (cnt = 0; stack[cnt]; cnt++) {
+	}
+	syms = blazesym_symbolize(ctx->symbolizer, ctx->cfgs, 2, (const uint64_t *)stack, cnt);
+	for (i = 0; stack[i]; i++) {
+		if (syms[i].valid)
+			if (syms[i].path && syms[i].path[0])
```

Comment:
it's my personal preference, but even though printf() below is a single statement and kernel code style allows to not have {} for such if/for/while statements, I think it looks bad if the body of if/for/while is more than a single line, so can you please add {} around multi-line statements? Thanks!

Reply:
Sure!  I personally prefer always to have braces.